test/cmd/newapp.sh

```diff
@@ -9,10 +9,18 @@ source "${OS_ROOT}/hack/util.sh"
 source "${OS_ROOT}/hack/cmd_util.sh"
 os::log::install_errexit
 
+# Cleanup cluster resources created by this test
+(
+  set +e
+  oc delete all,templates --all
+  exit 0
+) &>/dev/null
```

Comment:
Pushed an update to move the test json into fixtures.

On Tue, Feb 2, 2016 at 12:06 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> That's possible.  But you could also accidentally delete things by running
> oc delete all --all in normal use, or run rm -rf *.  At a certain level,
> running tests accepts a risk.
>
> On Tue, Feb 2, 2016 at 12:04 PM, Rodolfo Carvalho <
> notifications@github.com> wrote:
>
>> In test/cmd/newapp.sh
>> <https://github.com/openshift/origin/pull/6670#discussion_r51598881>:
>>
>> > @@ -9,10 +9,18 @@ source "${OS_ROOT}/hack/util.sh"
>> >  source "${OS_ROOT}/hack/cmd_util.sh"
>> >  os::log::install_errexit
>> >
>> > +# Cleanup cluster resources created by this test
>> > +(
>> > +  set +e
>> > +  oc delete all,templates --all
>> > +  exit 0
>> > +) &>/dev/null
>>
>> I could document that during development one could run test/cmd/*.sh
>> files to test against a running server.
>>
>> However, this also means that running a test might accidentally delete
>> resources you had in the server.
>>
>> Could we protect this clean up somehow? Say, only delete resources if an
>> environment variable OPENSHIFT_TEST_CMD_FORCE_CLEANUP is set.
>> Then, by default, running a test would not delete resources in whatever
>> namespace/project you happen to be.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openshift/origin/pull/6670/files#r51598881>.
>>
>
>